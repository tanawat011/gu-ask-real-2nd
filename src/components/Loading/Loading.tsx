import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import tw from 'twin.macro'

const TwContainer = tw.div`absolute top-1/2 left-1/2 flex h-full w-full -translate-x-1/2 -translate-y-1/2 cursor-not-allowed items-center justify-center rounded-md text-anti-flash-white`

export const Loading = () => {
  return (
    <TwContainer>
      <FontAwesomeIcon icon='spinner' spin />
    </TwContainer>
  )
}
