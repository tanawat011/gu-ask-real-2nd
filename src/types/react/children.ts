export type Children = { children: JSX.Element }
