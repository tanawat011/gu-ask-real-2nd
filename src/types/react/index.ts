export * from './children'
export * from './reactRouterDom'
