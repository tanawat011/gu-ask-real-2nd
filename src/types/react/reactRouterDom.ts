import type { Scroll } from 'types/common'

export type OutletContextProps = {
  scroll: Scroll
}
