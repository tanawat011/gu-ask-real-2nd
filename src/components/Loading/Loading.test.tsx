import { render } from '@testing-library/react'

import { Loading } from './Loading'

jest.mock('./Loading', () => jest.requireActual('./Loading'))

describe('<Loading />', () => {
  test('renders correctly', () => {
    expect.assertions(1)

    const { asFragment } = render(<Loading />)

    expect(asFragment()).toMatchSnapshot()
  })
})
