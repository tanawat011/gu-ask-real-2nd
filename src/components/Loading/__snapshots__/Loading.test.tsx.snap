// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders correctly 1`] = `
<DocumentFragment>
  .c1 {
  position: absolute;
  top: 50%;
  left: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  --tw-translate-x: -50%;
  -webkit-transform: translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  -ms-transform: translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  transform: translate(var(--tw-translate-x),var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  --tw-translate-y: -50%;
  cursor: not-allowed;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.375rem;
  --tw-text-opacity: 1;
  color: rgb(75 85 99 / var(--tw-text-opacity));
}

.dark .c0 {
  --tw-text-opacity: 1;
  color: rgb(249 250 251 / var(--tw-text-opacity));
}

<div
    class="c0 c1"
  >
    <svg>
      spinner
    </svg>
  </div>
</DocumentFragment>
`;
